src/api/Feed.js

```diff
@@ -95,6 +102,96 @@ const getItemWithPendingReply = <T: { replies?: Array<Comment> }>(item: T, reply
     return { replies: [...replies, reply], ...rest };
 };
 
+export const getParsedFileActivitiesResponse = (response?: { entries: FileActivity[] }) => {
+    if (!response || !response.entries || !response.entries.length) {
+        return { entries: [] };
+    }
+
+    const data = response.entries;
+    const parseReplies = (replies: Comment[]): Comment[] => {
+        const parsedReplies = [...replies];
+
+        return parsedReplies.map(reply => {
+            return { ...reply, tagged_message: reply.tagged_message || reply.message || '' };
+        });
+    };
+
+    const parsedData: Array<Object> = data
+        .map(item => {
+            if (!item.source) {
+                return null;
+            }
+
+            const source = { ...item.source };
+
+            switch (item.activity_type) {
+                case FILE_ACTIVITY_TYPE_TASK: {
+                    const taskItem = { ...source[FILE_ACTIVITY_TYPE_TASK] };
+                    // UAA follows a lowercased enum naming convention, convert to uppercase to align with task api
+                    if (taskItem.assigned_to?.entries) {
+                        const assignedToEntries = taskItem.assigned_to.entries.map(entry => {
+                            const assignedToEntry = { ...entry };
+
+                            assignedToEntry.role = entry.role.toUpperCase();
+                            assignedToEntry.status = entry.status.toUpperCase();
+
+                            return assignedToEntry;
+                        });
+                        // $FlowFixMe Converting to uppercase makes role and status a string, which is incompatible with string literal
```

Comment:
Not quite following this comment. Upper-casing shouldn't have an effect on the type.

Reply:
Worded poorly on my part, but Flow assumes that `role` and `status` is a string because it's using the `toUpperCase` method. `role` and `status` are supposed to be string literals based on the typing used, so Flow gives an error